### PR TITLE
Adds net-exporter chart

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/3711

```
$ helm list --tiller-namespace giantswarm
NAME                    	REVISION	UPDATED                 	STATUS  	CHART                                                       	NAMESPACE
chart-operator          	1       	Wed Aug 15 14:53:32 2018	DEPLOYED	chart-operator-chart-0.2.2                                  	giantswarm
external-dns            	1       	Wed Aug 15 14:58:18 2018	DEPLOYED	kubernetes-external-dns-chart-0.1.0                         	kube-system
kube-state-metrics      	1       	Wed Aug 15 14:57:43 2018	DEPLOYED	kubernetes-kube-state-metrics-chart-0.1.7                   	kube-system
net-exporter            	1       	Wed Aug 15 14:58:11 2018	DEPLOYED	net-exporter-chart-1.0.0-befd30be628347c9380ccdf6230c5a88...	kube-system
nginx-ingress-controller	1       	Wed Aug 15 14:58:14 2018	DEPLOYED	kubernetes-nginx-ingress-controller-chart-0.2.12            	kube-system
node-exporter           	1       	Wed Aug 15 14:57:59 2018	DEPLOYED	kubernetes-node-exporter-chart-0.1.8                        	kube-system
```

```
$ kubectl -n=kube-system get ds,pod -o wide -l=app=net-exporter
NAME                                DESIRED   CURRENT   READY     UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE       CONTAINERS     IMAGES                                                                     SELECTOR
daemonset.extensions/net-exporter   4         4         4         4            4           <none>          1h        net-exporter   quay.io/giantswarm/net-exporter:befd30be628347c9380ccdf6230c5a8865ea0688   app=net-exporter

NAME                     READY     STATUS    RESTARTS   AGE       IP           NODE
pod/net-exporter-cjhdf   1/1       Running   0          1h        10.1.130.5   4bims-worker-000003
pod/net-exporter-cp442   1/1       Running   0          1h        10.1.128.2   4bims-master-000000
pod/net-exporter-sdwvt   1/1       Running   0          1h        10.1.131.4   4bims-worker-000000
pod/net-exporter-zdhnf   1/1       Running   0          1h        10.1.129.3   4bims-worker-000002
```

```
$ curl -Ss 10.1.130.5:8000/metrics | grep network | head -n 10
# HELP network_error_total Total number of network dial errors.
# TYPE network_error_total counter
network_error_total 3
# HELP network_latency_seconds Histogram of latency of network dials.
# TYPE network_latency_seconds histogram
network_latency_seconds_bucket{host="net-exporter:8000",le="0.001"} 1
network_latency_seconds_bucket{host="net-exporter:8000",le="0.002"} 1
network_latency_seconds_bucket{host="net-exporter:8000",le="0.004"} 2
network_latency_seconds_bucket{host="net-exporter:8000",le="0.008"} 2
network_latency_seconds_bucket{host="net-exporter:8000",le="0.016"} 2
```

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
Looks like some test failure after merging master. Taking care. Shouldn't massively affect reviewability.

**Comment by JosephSalisbury** [comment]
To check in the morning:
```
$ kubectl -n=kube-system logs -f net-exporter-cjhdf
{"caller":"github.com/giantswarm/net-exporter/network/network.go:133","level":"error","message":"could not get endpoints","stack":"\u0026errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, ListMeta:v1.ListMeta{SelfLink:\"\", ResourceVersion:\"\", Continue:\"\"}, Status:\"Failure\", Message:\"endpoints \\\"net-exporter\\\" is forbidden: User \\\"system:serviceaccount:kube-system:net-exporter\\\" cannot get endpoints in the namespace \\\"monitoring\\\"\", Reason:\"Forbidden\", Details:(*v1.StatusDetails)(0xc420145020), Code:403}}","time":"2018-08-15T15:10:33.23999+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:133","level":"error","message":"could not get endpoints","stack":"\u0026errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, ListMeta:v1.ListMeta{SelfLink:\"\", ResourceVersion:\"\", Continue:\"\"}, Status:\"Failure\", Message:\"endpoints \\\"net-exporter\\\" is forbidden: User \\\"system:serviceaccount:kube-system:net-exporter\\\" cannot get endpoints in the namespace \\\"monitoring\\\"\", Reason:\"Forbidden\", Details:(*v1.StatusDetails)(0xc42008e7e0), Code:403}}","time":"2018-08-15T15:10:38.931354+00:00"}
{"caller":"github.com/giantswarm/net-exporter/network/network.go:133","level":"error","message":"could not get endpoints","stack":"\u0026errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:\"\", APIVersion:\"\"}, ListMeta:v1.ListMeta{SelfLink:\"\", ResourceVersion:\"\", Continue:\"\"}, Status:\"Failure\", Message:\"endpoints \\\"net-exporter\\\" is forbidden: User \\\"system:serviceaccount:kube-system:net-exporter\\\" cannot get endpoints in the namespace \\\"monitoring\\\"\", Reason:\"Forbidden\", Details:(*v1.StatusDetails)(0xc420367c80), Code:403}}","time":"2018-08-15T15:10:42.431803+00:00"}
```
looks like RBAC

**Comment by rossf7** [comment]
On the RBAC issue what is creating the `monitoring` namespace in the guest cluster?

We use the namespace resource in cluster-operator to create the giantswarm namespace. Do we need to extend it to create monitoring? If so should it be `giantswarm-monitoring`? To avoid naming clashes with customer workloads.

**Comment by JosephSalisbury** [comment]
We have the namespace `monitoring` set in the code itself, so when net-exporter is accessing the Kubernetes API to get the endpoints for the service, it's not allowed by RBAC. Should be fine to fix.

**Comment by JosephSalisbury** [comment]
I think installing into the `kube-system` namespace is still the correct approach, I don't think we need any more namespaces currently.

File: pkg/v6/configmap/desired.go [hunk 1]
```diff
@@ -44,12 +45,17 @@ type Image struct {
 	Registry string `json:"registry"`
 }
 
+type NetExporter struct {
+	Namespace string `json:"namespace"`
+}
```
[the remark this answers is not in the document]
Comment (JosephSalisbury) [reply pkg/v6/configmap/desired.go]: Okay cool, I'll put it here in this PR, and make sure we have an issue for sorting it later.

**Comment by JosephSalisbury** [comment]
@rossf7 having a release channel and retagging to the china registry is really gonna blow up the scope here - this isn't a third party image we're managing. Thoughts?

My gut feeling is to go with this, and refactor it as part of improving the china registry / chart-operator on hosts stories.

**Pull request merged**